File: WinForms/VB/TreeMapMainDemo/TutorialControl.vb
Imports System.Collections.Generic
Imports System.Drawing
Imports System.IO
Imports System.Windows.Forms
Imports DevExpress.Drawing
Imports DevExpress.DXperience.Demos
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraEditors
Imports DevExpress.XtraLayout
Imports DevExpress.XtraPrinting.Native

Namespace DevExpress.XtraTreeMap.Demos

    Public Partial Class TreeMapTutorialControl
        Inherits TutorialControlBase
        Implements IDXMenuManager

        Private ReadOnly imageFormats As Dictionary(Of String, DXImageFormat) = New Dictionary(Of String, DXImageFormat)() From {{"bmp", DXImageFormat.Bmp}, {"jpeg", DXImageFormat.Jpeg}, {"gif", DXImageFormat.Gif}, {"tiff", DXImageFormat.Tiff}, {"png", DXImageFormat.Png}}

        Public Sub New()
            AutoMergeRibbon = True
            SuspendLayout()
            Name = "TreeTreeMapTutorialControl"
            Size = New Size(500, 500)
            ResumeLayout(False)
        End Sub

        Public Overrides Property AutoMergeRibbon As Boolean
            Get
                Return True
            End Get

            Set(ByVal value As Boolean)
            End Set
        End Property

        Public Property MenuManager As IDXMenuManager

        Public Overridable ReadOnly Property ActiveDemoModule As Control
            Get
                Return Nothing
            End Get
        End Property

        Private ReadOnly Property ComponentExportable As IComponentExportable
            Get
                Return TryCast(ActiveDemoModule, IComponentExportable)
            End Get
        End Property

        Private ReadOnly Property ComponentPrintable As IComponentPrintable
            Get
                Return TryCast(ActiveDemoModule, IComponentPrintable)
            End Get
        End Property

        Private Function Clone(ByVal newForm As Form) As IDXMenuManager Implements IDXMenuManager.Clone
            Return Me
        End Function

        Private Sub DisposeManager() Implements IDXMenuManager.DisposeManager
        End Sub

        Private Sub ShowPopupMenu(ByVal menu As DXPopupMenu, ByVal control As Control, ByVal pos As Point) Implements IDXMenuManager.ShowPopupMenu
            MenuManagerHelper.ShowMenu(menu, LookAndFeel, MenuManager, control, pos)
        End Sub

        Private Function GetImageFormatsFilter() As String
            Dim result As String = String.Empty
            For Each format As String In imageFormats.Keys
                result += String.Format("{0} Image (*.{1})|*.{1}|", format.ToUpper(), format.ToLower())
            Next

            Return result.TrimEnd("|"c)
        End Function

        Protected Sub InitializeControls(ByVal sunburstControl As SunburstControl, ByVal colorizerComboBoxEdit As ComboBoxEdit, ByVal varyColorInGroupLCI As LayoutControlItem, ByVal modeLCI As LayoutControlItem)
            Call DemoHelpersContainer.AddControl(sunburstControl, colorizerComboBoxEdit, varyColorInGroupLCI, modeLCI)
        End Sub

        Protected Sub InitializeControls(ByVal sunburstControl As SunburstControl)
            Call DemoHelpersContainer.AddControl(sunburstControl)
        End Sub

        Protected Overrides Sub AllowExport()
            EnabledPrintExportActions(True, True, ExportFormats.PDF Or ExportFormats.MHT Or ExportFormats.RTF Or ExportFormats.XLS Or ExportFormats.XLSX Or ExportFormats.Image, False)
        End Sub

        Protected Overrides Sub ExportToCore(ByVal filename As String, ByVal ext As String)
            If ComponentExportable IsNot Nothing Then
                Dim currentCursor As Cursor = Cursor.Current
                Cursor.Current = Cursors.WaitCursor
                If Equals(ext, "rtf") Then
                    ComponentExportable.ExportToRtf(filename)
                ElseIf Equals(ext, "pdf") Then
                    ComponentExportable.ExportToPdf(filename)
                ElseIf Equals(ext, "mht") Then
                    ComponentExportable.ExportToMht(filename)
                ElseIf Equals(ext, "xls") Then
                    ComponentExportable.ExportToXls(filename)
                ElseIf Equals(ext, "xlsx") Then
                    ComponentExportable.ExportToXlsx(filename)
                End If

                Cursor.Current = currentCursor
            End If
        End Sub

        Protected Overrides Sub ExportToImage()
            If ComponentExportable Is Nothing Then Return
            Dim fileName As String = MainFormHelper.GetFileName("", GetImageFormatsFilter())
            If Not String.IsNullOrEmpty(fileName) Then
                If ComponentExportable IsNot Nothing Then ComponentExportable.ExportToImage(fileName, imageFormats(Path.GetExtension(fileName).TrimStart("."c).ToLower()))
            End If
        End Sub

        Protected Overrides Sub ExportToPDF()
            ExportTo("pdf", "PDF document (*.pdf)|*.pdf")
        End Sub

        Protected Overrides Sub ExportToHTML()
            ExportTo("html", "HTML document (*.html)|*.html")
        End Sub

        Protected Overrides Sub ExportToMHT()
            ExportTo("mht", "MHT document (*.mht)|*.mht")
        End Sub

        Protected Overrides Sub ExportToXLS()
            ExportTo("xls", "XLS document (*.xls)|*.xls")
        End Sub

        Protected Overrides Sub ExportToXLSX()
            ExportTo("xlsx", "XLSX document (*.xlsx)|*.xlsx")
        End Sub

        Protected Overrides Sub ExportToRTF()
            ExportTo("rtf", "RTF document (*.rtf)|*.rtf")
        End Sub

        Protected Overrides Sub ExportToText()
            ExportTo("txt", "Text document (*.txt)|*.txt")
        End Sub

        Protected Overrides Sub PrintPreview()
            If ComponentPrintable IsNot Nothing Then ComponentPrintable.ShowPrintPreview()
        End Sub

        Protected Overrides Sub Print()
            If ComponentPrintable IsNot Nothing Then ComponentPrintable.Print()
        End Sub
    End Class
End Namespace

File: WinForms/CS/TreeMapMainDemo/TutorialControl.cs
using System.Collections.Generic;
using System.Drawing;
using System.IO;
using System.Windows.Forms;
using DevExpress.Drawing;
using DevExpress.DXperience.Demos;
using DevExpress.Utils.Menu;
using DevExpress.XtraEditors;
using DevExpress.XtraLayout;
using DevExpress.XtraPrinting.Native;

namespace DevExpress.XtraTreeMap.Demos {
    public partial class TreeMapTutorialControl : TutorialControlBase, IDXMenuManager {
        readonly Dictionary<string, DXImageFormat> imageFormats = new Dictionary<string, DXImageFormat>() {
            { "bmp", DXImageFormat.Bmp },
            { "jpeg", DXImageFormat.Jpeg },
            { "gif", DXImageFormat.Gif },
            { "tiff", DXImageFormat.Tiff },
            { "png", DXImageFormat.Png }
        };
        public TreeMapTutorialControl() {
            AutoMergeRibbon = true;
            SuspendLayout();
            Name = "TreeTreeMapTutorialControl";
            Size = new Size(500, 500);
            ResumeLayout(false);
        }

        public override bool AutoMergeRibbon { get { return true; } set { } }
        public IDXMenuManager MenuManager { get; set; }
        public bool ShouldSerializeMenuManager() {
            return MenuManager != null;
        }
        public void ResetMenuManager() {
            MenuManager = null;
        }
        public virtual Control ActiveDemoModule { get { return null; } }
        IComponentExportable ComponentExportable { get { return ActiveDemoModule as IComponentExportable; } }
        IComponentPrintable ComponentPrintable { get { return ActiveDemoModule as IComponentPrintable; } }

        IDXMenuManager IDXMenuManager.Clone(Form newForm) { return this; }
        void IDXMenuManager.DisposeManager() { }
        void IDXMenuManager.ShowPopupMenu(DXPopupMenu menu, Control control, Point pos) {
            MenuManagerHelper.ShowMenu(menu, LookAndFeel, MenuManager, control, pos);
        }
        string GetImageFormatsFilter() {
            string result = string.Empty;
            foreach(string format in imageFormats.Keys)
                result += string.Format("{0} Image (*.{1})|*.{1}|", format.ToUpper(), format.ToLower());
            return result.TrimEnd('|');
        }

        protected void InitializeControls(SunburstControl sunburstControl, ComboBoxEdit colorizerComboBoxEdit, LayoutControlItem varyColorInGroupLCI, LayoutControlItem modeLCI) {
            DemoHelpersContainer.AddControl(sunburstControl, colorizerComboBoxEdit, varyColorInGroupLCI, modeLCI);
        }
        protected void InitializeControls(SunburstControl sunburstControl) {
            DemoHelpersContainer.AddControl(sunburstControl);
        }
        protected override void AllowExport() {
            EnabledPrintExportActions(true, true,
                ExportFormats.PDF | ExportFormats.MHT | ExportFormats.RTF | ExportFormats.XLS | ExportFormats.XLSX | ExportFormats.Image,
                false);
        }
        protected override void ExportToCore(string filename, string ext) {
            if(ComponentExportable != null) {
                Cursor currentCursor = Cursor.Current;
                Cursor.Current = Cursors.WaitCursor;
                if(ext == "rtf")
                    ComponentExportable.ExportToRtf(filename);
                else if(ext == "pdf")
                    ComponentExportable.ExportToPdf(filename);
                else if(ext == "mht")
                    ComponentExportable.ExportToMht(filename);
                else if(ext == "xls")
                    ComponentExportable.ExportToXls(filename);
                else if(ext == "xlsx")
                    ComponentExportable.ExportToXlsx(filename);
                Cursor.Current = currentCursor;
            }
        }
        protected override void ExportToImage() {
            if(ComponentExportable == null)
                return;
            string fileName = MainFormHelper.GetFileName("", GetImageFormatsFilter());
            if(!string.IsNullOrEmpty(fileName)) {
                if(ComponentExportable != null)
                    ComponentExportable.ExportToImage(fileName, imageFormats[Path.GetExtension(fileName).TrimStart('.').ToLower()]);
            }
        }
        protected override void ExportToPDF() {
            ExportTo("pdf", "PDF document (*.pdf)|*.pdf");
        }
        protected override void ExportToHTML() {
            ExportTo("html", "HTML document (*.html)|*.html");
        }
        protected override void ExportToMHT() {
            ExportTo("mht", "MHT document (*.mht)|*.mht");
        }
        protected override void ExportToXLS() {
            ExportTo("xls", "XLS document (*.xls)|*.xls");
        }
        protected override void ExportToXLSX() {
            ExportTo("xlsx", "XLSX document (*.xlsx)|*.xlsx");
        }
        protected override void ExportToRTF() {
            ExportTo("rtf", "RTF document (*.rtf)|*.rtf");
        }
        protected override void ExportToText() {
            ExportTo("txt", "Text document (*.txt)|*.txt");
        }
        protected override void PrintPreview() {
            if(ComponentPrintable != null)
                ComponentPrintable.ShowPrintPreview();
        }
        protected override void Print() {
            if(ComponentPrintable != null)
                ComponentPrintable.Print();
        }
    }
}
